### DISCUSSION TITLE
table-striped on bootstrap doesn't work

### DISCUSSION THREAD

**Discussion by GermanZun** [rappasoft/laravel-livewire-tables#894] [Q&A]
Hi. table-striped doesn't work for me on bootstrap. Even on the demo sample doesn't work and i can see that table-striped class is aplied to the table. How can i enable it? Thanks!

https://tables.laravel-boilerplate.com/bootstrap-5

![Screenshot from 2022-08-29 12-13-29](https://user-images.githubusercontent.com/12190841/187246884-4a3c2935-c287-4685-a873-10c9352c97b1.png)

**Comment by rappasoft** [comment]
You're right, I'm not sure why it's not applying I'll have to look into it.

**Comment by GermanZun** [comment]
I think its because the hidden rows. Bootstrap applies striped based on odd even, including hidden rows. If i delete the hidden rows (or add a new hidden for each hidden row) it works.

![image](https://user-images.githubusercontent.com/12190841/188901060-68643846-8e8c-41f8-a334-71718cbe0868.png)

**Comment by rappasoft** [comment]
Ill be honest i'm not sure how to fix this one because of the hidden rows.

**Comment by dave2084** [comment]
I fixed this by adding a class to the table where it would have the hidden rows and then added some sass to override the bootstrap 5 css when that class was present:

In config I added:
```
->setTableAttributes([
    'class' => 'table-striped-collapse',
])
```

The sass:
```
.table-striped {
    &.table-striped-collapse {
        > tbody > {
            tr {
                &:nth-of-type(odd) > * {
                    --bs-table-accent-bg: inherit;
                    color: inherit;
                }

                &:nth-of-type(4n-3),
                &:nth-of-type(4n-2) {
                    > * {
                        --bs-table-accent-bg: var(--bs-table-striped-bg);
                        color: var(--bs-table-striped-color);
                    }
                }
            }
        }
    }
}
```

**Reply by dave2084** [reply]
Also, if you are using the table-hover class you'll need the following sass:

```
.table {
    &.table-striped {
        &.table-striped-collapse {
            > tbody > tr {
                &:nth-of-type(odd) > * {
                    --bs-table-accent-bg: inherit;
                    color: inherit;
                }

                &:nth-of-type(4n-3),
                &:nth-of-type(4n-2) {
                    > * {
                        --bs-table-accent-bg: var(--bs-table-striped-bg);
                        color: var(--bs-table-striped-color);
                    }
                }
            }

            &.table-hover {
                > tbody > tr:hover > * {
                    --bs-table-accent-bg: var(--bs-table-hover-bg);
                    color: var(--bs-table-hover-color);
                
                }
            }
        }
    }
}
```
